app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt

```diff
@@ -1,18 +1,96 @@
+import kotlinx.android.synthetic.main.layout_air_quality.*
+import kotlinx.android.synthetic.main.layout_air_quality.progressAirQuality
+import kotlinx.android.synthetic.main.layout_temperature.*
+import kotlinx.android.synthetic.main.layout_weather_detail.*
+import java.text.SimpleDateFormat
+import java.util.*
 
-class CurrentWeatherFragment : BaseFragment() {
+class CurrentWeatherFragment : BaseFragment(), CurrentWeatherForecastContact.View {
+    private var presenter: CurrentWeatherForecastContact.Presenter? = null
 
     override val layoutResource get() = R.layout.fragment_current_weather
 
     override fun setupViews() {
+        showCurrentTime()
     }
 
     override fun setupData() {
+        presenter = CurrentWeatherPresenter(
+            this,
+            CurrentWeatherRepository.getInstance(CurrentWeatherRemoteDataSource.getInstance()),
+            AQIRepository.getInstance(AQIRemoteDataSource.getInstance()),
+            CurrentCityRepository.getInstance(
+                CurrentCityLocalDataSource.getInstance(
+                    SharePreferenceHelper.getInstance(context)
+                )
+            )
+        )
     }
 
     override fun initActions() {
+        presenter?.getCurrentLocation()
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun showCurrentWeatherForecast(weather: CurrentWeather) {
+        textCurrentTemper.text = weather.currentTemp.currentTemp.toInt().toString()
+        textDescription.text = weather.currentWeather.description.capitalize()
+        textTemperatureData.text = formatString(
+            weather.currentTemp.min.toInt().toString(), weather.currentTemp.max.toInt()
+                .toString()
+        )
+        textWind.text =
+            weather.wind.speed.toInt().toString() + context?.resources?.getString(R.string.text_km)
+        textWindGusts.text =
+            weather.wind.degree.toInt().toString() + context?.resources?.getString(R.string.text_km)
+        textWindDegree.text =
+            weather.currentTemp.humidity.toInt()
+                .toString() + context?.resources?.getString(R.string.text_percent)
+
+    }
+
+    override fun showAQIForecast(aqi: AQI) {
+        textAqi.text = aqi.aqi.toString()
+        progressAirQuality.progress = aqi.aqi
+        textMeasure.text =
+            context?.let { AirPollutionUtils.getAirPollutionTitle(it, aqi.aqi) }
+        textAirDescription.text =
+            context?.let { AirPollutionUtils.getAirPollutionDescription(it, aqi.aqi) }
+    }
+
+    override fun updateLocation(lat: String, lon: String) {
+        presenter?.getCurrentWeatherForecast(lat, lon)
+        presenter?.getAQI(lat, lon)
+    }
+
+    override fun showMessage(data: Any) {
+    }
+
+    private fun formatString(min: String, max: String) =
+        min + " " + context?.resources?.getString(R.string.text_degree) + " - " + max + " " + context?.resources?.getString(
+            R.string.text_degree
+        )
```

Comment:
theo minfh thì k nên viết gọn quá như kia tường minh ra chút . Không phải lúc nào ngắn gọn cx tốt đâu, viết code dễ đọc, dễ hiểu ms tốt .
```suggestion
   private fun formatStringTemp(min: String, max: String)  : String {
        val tempDescription = ""
        context?.resources?.let {
                 tempDescription = " min $it.getString(R.string.text_degree)  - max $it.getString(R.string.text_degree )
                 }
      return tempDescription
}
```